File: tree.py
from __future__ import annotations

from node import Node

class JsonTree:
    """Tree of linked lists that map out the JSON data"""

    def __init__(self, json_dict: 'JSONDict'):
        self.json_dict = json_dict
        self.map_tree(self.json_dict)

    def map_tree(self, json_dict):
        """Map the entire JSON tree and get access to leaf nodes"""
        self.leaf_nodes = []
        self.root_node = Node(json_data=json_dict, tree=self)

File: javascript/tree.js
let Node = require('./node.js')

class Tree {
    constructor(jsonDict) {
        this.jsonDict = jsonDict;
        this._mapTree(this.jsonDict);
    }

    _mapTree(jsonDict) {
        //Get edges to all Node's inside the Tree
        this.leafNodes = [];
        this.rootNode = new Node(jsonDict, this);
    }
}

module.exports = Tree